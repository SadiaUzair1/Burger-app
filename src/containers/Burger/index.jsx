import { BurgerIngredients, RenderMoreLessButton } from 'containers'

export const Burger = () => (
  <div>
    <div>
      <img src={'/images/upperbun.png'} alt={''} />
    </div>
    <main className={'container fitting'}>
      <BurgerIngredients />
      <img src={'/images/lowerbun.png'} alt={''} />
    </main>
    <div className={'container btn btn-warning mt-5'}>
      <RenderMoreLessButton />
    </div>
  </div>
)
