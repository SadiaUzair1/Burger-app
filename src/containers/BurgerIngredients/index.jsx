import { useSelector } from 'react-redux'

import { Bacon, Cheese, Lettuce, Meat } from 'components'

export const BurgerIngredients = () => {
  const type = ['lettuce', 'bacon', 'cheese', 'meat']
  const ingredient = useSelector(state => state.ingredient)

  const renderIngredients = type =>
    Array.from({ length: ingredient[type] }).map(() => {
      if (type === 'lettuce') {
        return <Lettuce />
      } else if (type === 'bacon') {
        return <Bacon />
      } else if (type === 'cheese') {
        return <Cheese />
      } else if (type === 'meat') {
        return <Meat />
      }
    })

  return (
    <div>
      {Array.from({ length: type.length }, (v, i) => (
        <div key={i}>{renderIngredients(type[i])}</div>
      ))}
    </div>
  )
}
