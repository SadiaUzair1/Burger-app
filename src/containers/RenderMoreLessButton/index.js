import { useDispatch, useSelector } from 'react-redux'

import Button from 'containers/RenderMoreLessButton/ButtonStyledComponent'
import { handleChange } from 'helper'

export const RenderMoreLessButton = () => {
  const amountAndIngredient = useSelector(state => state)
  const dispatch = useDispatch()
  const type = ['lettuce', 'bacon', 'cheese', 'meat']

  return (
    <div>
      <p className={'font-weight-bold'}>
        Current Price: ${amountAndIngredient.ingredient.amount.toFixed(2)}
      </p>
      {Array.from({ length: 4 }, (v, i) => (
        <div key={i}>
          <p className={'font-weight-bold'}>{type[i].toUpperCase()}</p>
          <button
            data-test={i + '-more'}
            onClick={() => handleChange(type[i], 'inc', amountAndIngredient, dispatch)}
            className={'text-white m-2'}
          >
            More
          </button>
          <button
            data-test={i + '-less'}
            onClick={() => handleChange(type[i], 'dec', amountAndIngredient, dispatch)}
            className={'text-white m-2'}
          >
            Less
          </button>
        </div>
      ))}
    </div>
  )
}
