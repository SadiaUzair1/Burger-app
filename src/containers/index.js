export { Burger } from 'containers/Burger'
export { BurgerIngredients } from 'containers/BurgerIngredients'
export { RenderMoreLessButton } from 'containers/RenderMoreLessButton'
