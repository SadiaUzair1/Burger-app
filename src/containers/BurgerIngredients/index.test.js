import { BurgerIngredients } from 'containers'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'

let wrapper
const setup = store =>
  shallow(
    <Provider store={store}>
      <BurgerIngredients />
    </Provider>
  ).childAt(0)

const mockStore = configureStore()
describe('rendering components', () => {
  const initialState = {
    ingredients: {
      lettuce: 1,
      bacon: 0,
      cheese: 0,
      meat: 0,
      amount: 3.0
    }
  }

  beforeEach(() => {
    const store = mockStore(initialState)
    wrapper = setup(store)
  })

  it(' burger ingredient should render with ingrient states', () => {
    expect(wrapper.length).toEqual(1)
  })

  it('store initial state should match', () => {
    const store = mockStore(initialState)
    const initialStates = {
      lettuce: 1,
      bacon: 0,
      cheese: 0,
      meat: 0,
      amount: 3.0
    }
    const state = store.getState()
    expect(state.ingredients).toEqual(initialStates)
  })

  it('burger ingredient should render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
