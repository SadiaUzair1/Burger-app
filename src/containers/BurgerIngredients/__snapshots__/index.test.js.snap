// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components burger ingredient should render 1`] = `<BurgerIngredients />`;
